libs/components/lists/testing/src/modules/filter/filter-inline-harness.ts

```diff
@@ -0,0 +1,57 @@
+import { HarnessPredicate } from '@angular/cdk/testing';
+import { SkyComponentHarness } from '@skyux/core/testing';
+
+import { SkyFilterInlineHarnessFilters } from './filter-inline-harness-filters';
+import { SkyFilterInlineItemHarness } from './filter-inline-item-harness';
+import { SkyFilterInlineItemHarnessFilters } from './filter-inline-item-harness-filters';
+
+/**
+ * Harness to interact with a filter inline component in tests.
+ */
+export class SkyFilterInlineHarness extends SkyComponentHarness {
+  /**
+   * @internal
+   */
+  public static hostSelector = 'sky-filter-inline';
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a
+   * `SkyFilterInlineHarness` that meets certain criteria.
+   */
+  public static with(
+    filters: SkyFilterInlineHarnessFilters,
+  ): HarnessPredicate<SkyFilterInlineHarness> {
+    return SkyFilterInlineHarness.getDataSkyIdPredicate(filters);
+  }
+
+  /**
+   * Gets a harness for a specific toolbar item that meets certain criteria.
+   */
+  public async getItem(
+    filter: SkyFilterInlineItemHarnessFilters,
+  ): Promise<SkyFilterInlineItemHarness> {
+    return await this.locatorFor(SkyFilterInlineItemHarness.with(filter))();
+  }
+
+  /**
+   * Gets an array of all toolbar items.
+   */
+  public async getItems(
+    filters?: SkyFilterInlineItemHarnessFilters,
+  ): Promise<SkyFilterInlineItemHarness[]> {
+    const items = await this.locatorForAll(
+      SkyFilterInlineItemHarness.with(filters || {}),
+    )();
+
+    if (items.length === 0) {
```

Comment:
im on the fence about this error implementation. bc if `locatorForAll` doesnt find anything it would return an empty array, there is an odd off chance that is expected behavior so sending an error to consumers doesnt feel like the right way to implement this

Reply:
from conversation with corey: in situations where consumers use filters to get specific child harnesses and those dont exist and we are using `locatorForAll` which when nothing is found returns an empty array and does not error, we want to error out to unify the experience between `locatorFor` and `locatorForAll` with filters. so keep this error handling.